src/org/skypro/skyshop/searchEngine/SearchEngine.java

```diff
@@ -48,6 +49,49 @@ public void printSearchResults(Searchable[] results) {
             }
         }
     }
+// Дз -4
+    public Searchable findBestMatch(String search) throws BestResultNotFound {
```

Comment:
исключения лучше называть с окончанием на Exception
```suggestion
    public Searchable findBestMatch(String search) throws BestResultNotFoundException {
```